### ISSUE TITLE
Strange clone layer behaviour

### ISSUE THREAD

**Issue by waxfingers** [aferrero2707/PhotoFlow#135]
Small thing, but in a layer stack I was using a couple curve's layers with a clone (linked to a buffer), a gradient and a curve. In one of the curve's layer ( _curves OUT_ in the attached pfp) behaviour was normal, on the second **_curves IN_**, clone layer skips order ( clone on the bottom, gradient and finally curves) and "insists" to stay on top 

[missbehaviour_clone.pfp.zip](https://github.com/aferrero2707/PhotoFlow/files/733711/missbehaviour_clone.pfp.zip)

**Comment by aferrero2707** [comment]
Strange, I loaded your preset and I have been able to drag the clone layer at the bottom of the mask stack without problems:

<img width="284" alt="screen shot 2017-01-31 at 15 18 21" src="https://cloud.githubusercontent.com/assets/6229237/22468595/232f0316-e7c9-11e6-9a76-a9ed76553861.png">

Also, did you know that you can apply a curve to a gradient directly within the gradient tool? Here is an example:

<img width="286" alt="screen shot 2017-01-31 at 15 21 50" src="https://cloud.githubusercontent.com/assets/6229237/22468617/3f6c2b26-e7c9-11e6-9cde-9da59154b4a0.png">

**Comment by waxfingers** [comment]
> Also, did you know that you can apply a curve to a gradient directly within the gradient tool? Here is an example:

I didn't 😃, is it like a luma mask inside the gradient?   
  
![clone_missbehaviour](https://cloud.githubusercontent.com/assets/13238340/22471650/e110ad2a-e7ca-11e6-95cd-979f9d719e7f.gif)

**Comment by aferrero2707** [comment]
You are right! If one of the other layers is moved ABOVE the clone, the configuration is kept. But if the clone (or wathever other layer) is moved at the bottom, the configuration is not kept next time the mask is opened... I will investigate.
